automol/util/dict_/_dict_.py

```diff
@@ -1,86 +1,93 @@
-""" Helper functions for working with Python dictionaries
-"""
+"""Helper functions for working with Python dictionaries."""
 
 import itertools
 from copy import deepcopy
 from typing import Any
 
 import numpy
+from _collections_abc import Callable
 
-# for dictionaries do dict [obj, tuple [obj, ...]]
-def invert(dct):
-    """Transposes the keys and values in a dictionary
 
-    :param dct: dictionary to transpose
-    :tupe dct: dict
-    :rtype: dict
+# for dictionaries do dict[object,object] [obj, tuple [obj, ...]]
+def invert(dct: dict[object, object]) -> dict[object, object]:
+    """Transposes the keys and values in a dictionary.
+
+    :param dct: Dictionary to transpose
+    :return: Transposed dictionary
     """
     return {val: key for key, val in dct.items()}
 
 
-def empty_if_none(obj):
+def empty_if_none(obj: object):
     """Returns an empty dictionary if input object is None,
     otherwise return the object.
-
-    :param obj: generic object
-    :type obj: any
-    """
+    :param obj: Generic object
+    :return: Object of none if none.
+    """  # noqa: D401
     return {} if obj is None else obj
 
 
-def compose(dct1, dct2):
-    """Get the composition of `dct1` with `dct2`
+def compose(
+    dct1: dict[object, object],
+    dct2: dict[object, object],
+) -> dict[object, object]:
+    """Get the composition of `dct1` with `dct2`.
 
     That is, dct[k] = dct1[dct2[k]]
 
     :param dct1: The first dictionary
-    :type dct1: dict
     :param dct2: The second dictionary
-    :type dct2: dict
     :returns: The composed dictionary
-    :rtype: dict
     """
     return {k2: dct1[v2] for k2, v2 in dct2.items()}
 
 
-def by_key(dct, keys, fill=True, fill_val=None):
-    """dictionary on a set of keys, filling missing entries
+def by_key(
+    dct: dict[object, object],
+    keys,
+    fill: bool = True,
+    fill_val: bool | None = None,
```

Comment:
The type of `fill_val` should just be `object` (still with a default value of `None`).